src/Core/HubSpotBaseClient.cs

```diff
@@ -252,20 +253,22 @@ protected async Task<T> ListPropertiesAsync<T>(string absoluteUriPath) where T :
         /// <param name="httpMethod">HTTP method to use for the request</param>
         /// <param name="json">Optional json to send with the request</param>
         /// <param name="deserializeFunc">Func to handle deserialization of data when the request goes well</param>
-        /// <returns>A deserialized entity with data when things go well, exceptionns otherwise</returns>
+        /// <returns>A deserialized entity with data when things go well, exceptions otherwise</returns>
         protected async Task<T> SendRequestAsync<T>(string absoluteUriPath, HttpMethod httpMethod, string json,
             Func<string, T> deserializeFunc)
         {
-            var fullUrl = $"{HubSpotBaseUrl}{absoluteUriPath}"
-                .SetQueryParam("hapikey", _apiKey);
-            
+            var fullUrl = $"{HubSpotBaseUrl}{absoluteUriPath}";
+
             Logger.LogDebug("Full url: '{0}'", fullUrl);
 
             var request = new HttpRequestMessage
             {
                 Method = httpMethod,
                 RequestUri = new Uri(fullUrl)
             };
+
+            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", _apiToken);
+            
```

Comment:
I would like to make this a non-breaking change. I know that Hubspot will effectively stop allowing the apiKey from 30'th Nov, but since we're still in 0.x semver, the breaking change might be missed by people upgrading.

It seems like the new private app keys start with `pat-`, so I think that we could do a check on this and then either attach the `Authorization` header, or use the `SetQueryParam` to use the legacy auth mode. Which should then issue a `Warning` via the logger - that the legacy auth is used which will be removed Dec 2022.